.github/workflows/github-container-registry.yml

```diff
@@ -56,89 +52,242 @@ jobs:
         run: git diff --exit-code
 
   build-php-images:
-    name: PHP ${{ matrix.php }} Image
-    runs-on: ubuntu-latest
+    name: PHP ${{ matrix.php }} (${{ matrix.arch.type }}) Image
+    runs-on: ${{ matrix.arch.runner }}
     needs: [ check-for-changes ]
+    permissions:
+        contents: read
     strategy:
+      fail-fast: false
       matrix:
         php: [ '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3', '8.4' ]
+        arch:
+          - type: linux/amd64
+            runner: ubuntu-24.04
+          - type: linux/arm64
+            runner: ubuntu-24.04-arm
     env:
       PHP_VERSION: ${{ matrix.php }}
-      PHP_LATEST: '8.2'
-    permissions:
-      contents: read
+      ARCH_TYPE: ${{ matrix.arch.type }}
 
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: Prepare
+        run: |
+          platform="php-${PHP_VERSION}-${ARCH_TYPE}"
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 # 5.7.0
         with:
-          persist-credentials: false
+          images: ${{ vars.GHCR_PHP_IMAGE }}
+      - name: Login to Registry
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # 3.4.0
+        with:
+  
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
At first it was not obvious why this would actually work. How would `wileEcoyote` be able to authenticate using `GITHUB_TOKEN` if they do not belong to the organization. But after researching, it appears that the `username` is not used for authentication and it's handled by the token itself.

We should include an inline comment here briefly explaining that for clarity. Same below for the CLI images.